app/Components/TreeInventory/index.js

```diff
@@ -62,18 +85,43 @@ const TreeInventory = ({ navigation }) => {
 
     let pendingInventory = []
     let inCompleteInventory = []
+    let uploadedInventory = []
     if (allInventory) {
         pendingInventory = allInventory.filter(x => x.status == 'pending')
         inCompleteInventory = allInventory.filter(x => x.status == 'incomplete')
+        uploadedInventory = allInventory.filter(x => x.status == 'complete')
+    }
+
+    const onPressUploadNow = () => {
+        setIsLoaderShow(true)
+        uploadInventory().then((data) => {
+            initialState()
+            setIsLoaderShow(false)
+        }).catch((err) => {
+            setIsLoaderShow(false)
+        })
+    }
+
+    const renderLoaderModal = () => {
+        return (<Modal
+            transparent
+            visible={isLoaderShow}>
+            <View style={styles.dowloadModalContainer}>
+                <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+                    <Text style={{ padding: 30, backgroundColor: '#fff', borderRadius: 10 }}>Uploading .......</Text>
```

Comment:
Before showing this modal, add a check if user is logged In.
If user is not logged in, redirect to login screen; log them in; before uploading.

Reply:
Fixed